src/main/java/org/springframework/integration/kafka/listener/ConcurrentMessageListenerDispatcher.java

```diff
@@ -120,19 +120,17 @@ public void dispatch(KafkaMessage message) {
 
 	private void initializeAndStartDispatching() {
 		// allocate delegate instances index them
-		List<QueueingMessageListenerInvoker> delegateList = new ArrayList<QueueingMessageListenerInvoker>(consumers);
+		List<QueueingMessageListenerInvoker> delegateList = new ArrayList<>(consumers);
 		for (int i = 0; i < consumers; i++) {
-			QueueingMessageListenerInvoker queueingMessageListenerInvoker =
-					new QueueingMessageListenerInvoker(queueSize, offsetManager, delegateListener, errorHandler,
-							taskExecutor, autoCommitOnError);
+			QueueingMessageListenerInvoker queueingMessageListenerInvoker = new QueueingMessageListenerInvoker(
+					queueSize, offsetManager, delegateListener, errorHandler, taskExecutor, autoCommitOnError);
 			delegateList.add(queueingMessageListenerInvoker);
 		}
 
 		// evenly distribute partitions across delegates
 		delegates = Maps.mutable.of();
-		int i = 0;
 		for (Partition partition : partitions) {
-			delegates.put(partition, delegateList.get((i++) % consumers));
+			delegates.put(partition, delegateList.get((partitionAssignmentCounter.incrementAndGet()) % consumers));
```

Comment:
Was there some other purpose of changing it from a local int to an `AtomicInteger` ?